sdmmc driver support

Deepak-AlifSemi opened the pull request:
On Top of SAU and TGU Setup, SD Support patch is added.

Deepak-AlifSemi committed:
driver: sdhc: Enabled intel sdhc driver for alif ensemble soc family

Added SDMMC support in zephyr using existing intel_emmc_host driver
Added SDMMC Pinmux for SD_A peripheral of Ensemble
Added Customised generic section placement for DMA Buffers

Signed-off-by: Deepak Kumar <deepak@alifsemi.com>

Deepak-AlifSemi commented on drivers/sdhc/intel_emmc_host.c:
```diff
@@ -415,7 +422,8 @@ static int emmc_dma_init(const struct device *dev, struct sdhc_data *data, bool
 
 	if (IS_ENABLED(CONFIG_INTEL_EMMC_HOST_ADMA)) {
-		uint8_t *buff = data->data;
+		uint8_t *buff = (data->data);
```
resolved

Deepak-AlifSemi commented on drivers/sdhc/intel_emmc_host.c:
```diff
@@ -1235,6 +1267,15 @@ static int emmc_init(const struct device *dev)
+#if defined(CONFIG_PINCTRL)
+	int ret = pinctrl_apply_state(config->pincfg, PINCTRL_STATE_DEFAULT);
+
+	LOG_DBG("Applying Pinctrl state. ret - %d", ret);
```
resolved

Deepak-AlifSemi commented on dts/arm/alif/ensemble_rtss_common.dtsi:
```diff
@@ -1126,6 +1126,29 @@
 		};
 	};
 
+	sdhc: sdhc@48102000 {
```
removed blank lines

silesh-alifsemi reviewed (dismissed):
Please update the commit message for the first commit. You do not need to mention things that were discussed during the code review in the commit message. For eg. "Indentation and comment resolved." For this specific commit, I think we can mention the changes that we are introducing in this commit for the TGU/SAU initialization (moving the routines to `hal_alif`, adding the hook to` z_platform_init`  etc.)and how a user can modify the location and size of the NS region.

RupeshKumar-AlifSemi commented:
Please run the gitlint and checkpatch.. there are issues related to it

rajranjan-alifsemi commented on subsys/sd/sdmmc.c:
```diff
@@ -726,6 +746,12 @@ int sdmmc_card_init(struct sd_card *card)
+			LOG_ERR("Could not set SD blocklen to 512");
+			return ret;
+		}
+		card->block_size = 512;
```
remove hardcoding.

rajranjan-alifsemi commented on drivers/sdhc/Kconfig.intel:
```diff
@@ -13,6 +13,13 @@ config INTEL_EMMC_HOST
+	string
+	default ".bss"
+	help
+		SD Card internal Buffer Linker Section
```
indentation same as other configs in this file

Deepak-AlifSemi replied:
done

rajranjan-alifsemi commented on subsys/sd/Kconfig:
```diff
@@ -80,6 +81,12 @@ config SD_BUFFER_SIZE
+	string
+	default ".bss"
+	help
+		SD Card internal Buffer Linker Section
```
try to align as others in same file

Deepak-AlifSemi replied:
done

rajranjan-alifsemi commented on samples/subsys/fs/fs_sample/boards/alif_e7_dk_rtss_hp.conf:
```diff
@@ -0,0 +1,21 @@
+
+#FATFS
+CONFIG_FS_SAMPLE_CREATE_SOME_ENTRIES=y
+CONFIG_FS_FATFS_WINDOW_ALIGNMENT=512
```
default value is 512 only why explicit declaration

sudhir-alifsemi commented on subsys/fs/Kconfig.fatfs:
```diff
@@ -210,7 +210,7 @@ config FS_FATFS_MIN_SS
 config FS_FATFS_WINDOW_ALIGNMENT
 	int "Memory alignment for the member \"win\" in FATFS"
-	default 1
+	default 512
```
use the default

sudhir-alifsemi commented on subsys/sd/sd.c:
```diff
@@ -18,6 +18,10 @@
 
+static uint8_t Z_GENERIC_SECTION(CONFIG_SD_BUFFER_SECTION)
+		cmd_dma_buff[CONFIG_SD_BUFFER_SIZE]
+		__aligned(MAX(4, CONFIG_SDHC_BUFFER_ALIGNMENT));
```
set the buffer alignment from prj.conf

Deepak-AlifSemi replied:
done

sudhir-alifsemi commented on subsys/sd/sdmmc.c:
```diff
@@ -468,6 +485,9 @@ static int sdmmc_set_bus_speed(struct sd_card *card)
 		/* Change host bus speed */
 		card->bus_io.timing = timing;
 		card->bus_io.clock = card->switch_caps.uhs_max_dtr;
+#if defined(CONFIG_SOC_FAMILY_ENSEMBLE) || defined(CONFIG_SOC_FAMILY_BALLETTO)
```
this should have been passed as a flag (quirks) from dts to check the HW capability and set the clock speed

Deepak-AlifSemi commented on drivers/sdhc/intel_emmc_host.c:
```diff
@@ -653,10 +667,27 @@ static void update_cmd_response(const struct device *dev, struct sdhc_command *s
 
 		LOG_DBG("cmd resp: %x %x %x %x", resp0, resp1, resp2, resp3);
 
+#if defined(CONFIG_SOC_FAMILY_ENSEMBLE) || defined(CONFIG_SOC_FAMILY_BALLETTO)
```
done

Deepak-AlifSemi committed:
drivers: sdhc: intel_emmc_host: Added local_to_global conversion

Convert dtcm local address to global addres before submitting to DMA
for SD/MMC DMA accessibility.

Signed-off-by: Deepak Kumar <deepak@alifsemi.com>
